marathon_lb.py

```diff
@@ -1323,8 +1331,20 @@ def get_apps(marathon):
         service_ports = app['ports']
         for i in range(len(service_ports)):
             servicePort = service_ports[i]
+            # Find the container port tied to this service port
+            container_port = None
+            for port_mapping in app['container']['docker']['portMappings']:
+                if port_mapping['servicePort'] == servicePort:
+                    container_port = port_mapping['containerPort']
+                    break
+            if not container_port:
+               raise Exception("No matching container port found for service port %s" % servicePort) 
+
             service = MarathonService(
-                        appId, servicePort, get_health_check(app, i))
+                        appId, servicePort, container_port, get_health_check(app, i))
+
+            if 'HAPROXY_IP_PER_CONTAINER' in marathon_app.app['labels']:
```

Comment:
Perhaps use the same pattern that's used with `HAPROXY_{0}_REDIRECT_TO_HTTPS` and `HAPROXY_{0}_USE_HSTS`.